silence/silence.go

```diff
@@ -518,9 +518,6 @@ func matchesEmpty(m *pb.Matcher) bool {
 }
 
 func validateSilence(s *pb.Silence) error {
-	if s.Id == "" {
```

Comment:
I have removed checks for the `Id` and `UpdatedAt` fields from `validateSilence`. There are two reasons for this:

1. I have moved validation of silences to the start of the function before the `Id` and `UpdatedAt` fields are assigned. This is to fix the issue where an existing silence is expired but the new silence is not created.
2. These fields are supposed to be set in `silences.Set(*pb.Silence)`, not by the calling code. We assert that these are present within tests, not validation code that covers both internal and external fields.